src/LibraryUtilities.ts

```diff
@@ -295,3 +297,24 @@ export function searchItemResursive(items: ItemData[], text: string) {
         search(text, item);
     }
 }
+
+export function getHighlightedText(text: string, highlightedText: string) {
+    if (highlightedText.length == 0) {
+        return text;
+    }
+
+    var regex = new RegExp('' + highlightedText + '', 'gi');
+    var segments = text.split(regex);
+    var replacements = text.match(regex);
+    var spans = [];
```

Comment:
Can we use `let` instead of `var` here? Also for `spans` we need its type to be specified (if possible).